src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -98,6 +98,20 @@ bool ClientRequest::is_pg_op() const
     [](auto& op) { return ceph_osd_op_type_pg(op.op.op); });
 }
 
+auto ClientRequest::reply_op_error(const Ref<PG>& pg, int err)
+{
+  LOG_PREFIX(ClientRequest::reply_op_error);
+  DEBUGDPP("{}: replying with error {}", *pg, *this, err);
+  auto reply = crimson::make_message<MOSDOpReply>(
+    m.get(), err, pg->get_osdmap_epoch(),
+    m->get_flags() & (CEPH_OSD_FLAG_ACK|CEPH_OSD_FLAG_ONDISK),
+    !m->has_flag(CEPH_OSD_FLAG_RETURNVEC));
+  reply->set_reply_versions(eversion_t(), 0);
+  reply->set_op_returns(std::vector<pg_log_op_return_item_t>{});
+  // TODO: gate the crosscore sending
+  return get_foreign_connection().send_with_throttling(std::move(reply));
+}
+
```

Comment:
Is this a meanless change?

Reply:
No, `reply_op_error()` returns deduced type, so it has to be defined before used. So I moved its definition before `with_pg_int()`

Reply:
nit: I think that returning `seastar::future<>` is more readable and avoids the issue you encountered.